src/eradiate_plugins/bsdfs/ocean_legacy.cpp

```diff
@@ -0,0 +1,1381 @@
+     * @brief Evaluate the sun glint reflectance.
+     *
+     * Evaluates the sun glint reflectance at the given wavelength, incident and
+     * outgoing directions, and wind direction. The effects of wind speed and 
+     * chlorinity are precomputed in update(). The reflectance is computed using 
+     * the Cox-Munk distribution, the Fresnel equations, and relative tilt of 
+     * the oceanic surface.
+     *
+     * @param n_real The real part of the index of refraction of water.
+     * @param n_imag The imaginary part of the index of refraction of water.
+     * @param wi The incident direction of the light.
+     * @param wo The outgoing direction of the light.
+     * @param wind_direction The direction of the wind relative to the incident
+     * light.
+     * @return Float The sun glint reflectance.
+     */
+    Float eval_glint(const ScalarFloat &n_real, const ScalarFloat &n_imag,
+                     const Vector3f &wi, const Vector3f &wo,
+                     const ScalarFloat &wind_direction) const {
+        // Transform directions into azimuthal and zenithal angles
+        Float s_theta_i = dr::sqrt(1 - wi.z() * wi.z());
+        Float c_theta_i = wi.z();
+
+        Float s_theta_o = dr::sqrt(1 - wo.z() * wo.z());
+        Float c_theta_o = wo.z();
+
+        Float phi_i         = dr::atan2(wi.y(), wi.x());
+        Float phi_o         = dr::atan2(wo.y(), wo.x());
+        Float phi           = phi_i - phi_o;
+        auto [s_phi, c_phi] = dr::sincos(phi);
+
+        Float phi_w             = phi_i - wind_direction;
+        auto [s_phi_w, c_phi_w] = dr::sincos(phi_w);
+
+        return eval_sun_glint(n_real, n_imag, s_theta_i, c_theta_i, s_theta_o,
+                              c_theta_o, s_phi, c_phi, s_phi_w, c_phi_w);
+    }
+
+    /**
+     * @brief Evaluate the sun glint reflectance.
+     *
+     * Evaluates the sun glint reflectance at the given index of refraction, 
+     * incident and outgoing angles, relative azimuthal angle for 
+     * ingoing and outgoing directions, and relative azimuthal angle for 
+     * wind direction. The reflectance is computed using the Cox-Munk
+     * distribution, the Fresnel equations, and relative tilt of the 
+     * oceanic surface.
+     *
+     * @param n_real The real part of the index of refraction of water.
+     * @param n_imag The imaginary part of the index of refraction of water.
+     * @param s_i The sine of the incident zenith angle.
+     * @param c_i The cosine of the incident zenith angle.
+     * @param s_o The sine of the outgoing zenith angle.
+     * @param c_o The cosine of the outgoing zenith angle.
+     * @param s_phi The sine of the incident azimuthal angle.
+     * @param c_phi The cosine of the incident azimuthal angle.
+     * @param s_phi_w The sine of the relative azimuthal angle of the wind.
+     * @param c_phi_w The cosine of the relative azimuthal angle of the wind.
+     * @param invert_real Whether to invert the real part of the IOR.
```

Comment:
From my understanding, this is used for upwelling transmittance to take into account the fact that light is transmitted from water to air instead of air to water.